src/utils.cpp

```diff
@@ -1,34 +1,63 @@
-#include "utils.hpp"
-
-namespace utils {
-/**
- * Saves a JSON
- * @param json to be saved
- * @param path to save the json
- */
-void saveJson(const nlohmann::json& json, std::filesystem::path path)
-{
-    if (!std::filesystem::is_directory(path)) {
-        std::filesystem::create_directory(path);
-    }
-
-    time_t rawtime;
-    struct tm* timeinfo;
-
-    constexpr int bufferSize(80);
-    std::array<char, bufferSize> buffer;
-
-    time(&rawtime);
-    timeinfo = localtime(&rawtime);
-
-    strftime(buffer.data(), bufferSize, "%F-%H-%M-%S", timeinfo);
-
-    path /= buffer.data();
-    path += ".json";
-
-    std::ofstream file(path);
-    std::cout << "Wrote file: " << path << std::endl;
-    file << json;
-    file.close();
-}
-} // namespace utils
+#include "utils.hpp"
+
+namespace utils {
+/**
+ * Saves a JSON
+ * @param json to be saved
+ * @param path to save the json
+ */
+void saveJson(const nlohmann::json& json, std::filesystem::path path)
+{
+    std::filesystem::create_directory(path);
+
+    time_t rawtime;
+    struct tm* timeinfo;
+
+    constexpr int bufferSize(80);
+    std::array<char, bufferSize> buffer;
+
+    time(&rawtime);
+    timeinfo = localtime(&rawtime);
+
+    strftime(buffer.data(), bufferSize, "%F-%H-%M-%S", timeinfo);
+
+    path /= buffer.data();
+    path += ".json";
+
+    std::ofstream file(path);
+    std::cout << "Wrote file: " << path << std::endl;
+    file << json;
+    file.close();
+}
+
+// got it from https://stackoverflow.com/questions/5878775/how-to-find-and-replace-string
+void replaceAll(std::string& s, std::string const& toReplace, std::string const& replaceWith)
```

Comment:
I believe this is a better reference: https://thispointer.com/find-and-replace-all-occurrences-of-a-sub-string-in-c/

Check this options!